core/extension/gdextension_manager.cpp

```diff
@@ -236,6 +236,18 @@ void GDExtensionManager::load_extensions() {
 	}
 
 	OS::get_singleton()->load_platform_gdextensions();
+
+	if (p_init_func) {
+		Ref<GDExtension> libgodot;
+		libgodot.instantiate();
+		Error err = libgodot->initialize_extension_function(p_init_func, "lib_godot");
```

Comment:
Rather than passing around the `GDExtensionInitializationFunction p_init_func`, and finally getting around to calling it here, could we simply register the `GDExtensionInitializationFunction` into some list (a list assuming we could have multiple) and then loop over them here? (Again, this depends on if we want to allow multiple or not.)

Reply:
I would prefer to be able to use multiple godot gdextension symbols in the same libgodot binary as a user.

Reply:
As I mentioned above, we had support for it in an earlier version of the patch, we will resurrect it in the next version of this PR.

Reply:
Added a new load_function_extension() method exposed to GDExtension in the GDExtensionManager that is using the new function loader internally.